aospy/data_loader.py

```diff
@@ -0,0 +1,416 @@
+                      time_offset=None, **DataAttrs):
+        """Return a DataArray with requested variable, given time range,
+        and input interval.
+
+        This function automatically renames all grid attributes to match
+        aospy conventions.
+
+        Parameters
+        ----------
+        var : Var
+            aospy Var object
+        start_date : datetime.datetime
+            start date for interval
+        end_date : datetime.datetime
+            end date for interval
+        time_offset : dict
+            Option to add a time offset to the time coordinate to correct for
+            incorrect metadata.
+        **DataAttrs
+            Attributes needed to identify a unique set of files to load from
+
+        Returns
+        -------
+        da : DataArray
+             DataArray for the specified variable, date range, and interval in
+        """
+        file_set = self._generate_file_set(var=var, start_date=start_date,
+                                           end_date=end_date, **DataAttrs)
+        ds = self._load_data_from_disk(file_set)
+        ds = self._prep_time_data(ds)
+        ds = self.set_grid_attrs_as_coords(ds)  # Tested
+        da = self._sel_var(ds, var)  # Tested
+
+        # Apply correction before selecting time range.
+        # Note that time shifts are a property of a particular list of files
+        # NOT an entire DataLoader, so there either needs to be a way to
+        # specify those on a file list by file list basis. By default
+        # I think it is best to make the user be explicit about it in Calc.
+        # If one wants more automation, one can
+        # override _maybe_apply_time_shift
+        # in a new DataLoader and automate the offset based on
+        # DataAttrs like 'inst' for example
+        da = self._maybe_apply_time_shift(da, time_offset, **DataAttrs)
+        start_date_xarray = times.numpy_datetime_range_workaround(start_date)
+        end_date_xarray = start_date_xarray + (end_date - start_date)
+        return times.sel_time(da, np.datetime64(start_date_xarray),
+                              np.datetime64(end_date_xarray)).load()
+
+    @staticmethod
+    def _maybe_apply_time_shift(da, time_offset=None, **DataAttrs):
+        """Apply specified time shift to DataArray"""
+        if time_offset is not None:
+            time = times.apply_time_offset(da[internal_names.TIME_STR],
+                                           **time_offset)
+            da[internal_names.TIME_STR] = time
+        return da
+
+    @classmethod
+    def _load_data_from_disk(cls, file_set):
+        """Load a Dataset from a list of files, concatenating along time,
+        and rename all grid attributes to their aospy internal names.
+
+        Parameters
+        ----------
+        file_set : list
+            List of paths to files
+
+        Returns
+        -------
+        Dataset
+        """
+        io.dmget(file_set)
```

Comment:
I would like to avoid hardcoding in this GFDL-specific logic.  For now, can you just wrap this in another function called something like `apply_preload_user_commands` (that name could be better).  e.g.
```
def apply_preload_user_commands(file_set, cmd=io.dmget):
    """Call desired functions on file list before loading.

    For example, on the NOAA Geophysical Fluid Dynamics Laboratory computational cluster, data that is saved on their tape archive must be accessed via a `dmget` (or `hsmget`) command before being used.
    """
    if cmd is not None:
        cmd(file_set)
```
That way it will be easier to later enable users to specify this with their own needed system calls.